hydroflow/src/util/tcp.rs

```diff
@@ -83,60 +83,63 @@ pub async fn bind_tcp<T: 'static, Codec: 'static + Clone + Decoder + Encoder<T>>
 
     let bound_endpoint = listener.local_addr()?;
 
-    let (tx_egress, mut rx_egress) = unsync_channel(None);
-    let (tx_ingress, rx_ingress) = unsync_channel(None);
-
-    let clients = Rc::new(RefCell::new(HashMap::new()));
-
-    spawn_local({
-        let clients = clients.clone();
-
-        async move {
-            while let Some((payload, addr)) = rx_egress.next().await {
-                let client = clients.borrow_mut().remove(&addr);
-
-                if let Some(mut sender) = client {
-                    let _ = SinkExt::send(&mut sender, payload).await;
-                    clients.borrow_mut().insert(addr, sender);
-                }
-            }
-        }
-    });
+    let (send_egress, mut recv_egress) = unsync_channel::<(T, SocketAddr)>(None);
+    let (send_ingres, recv_ingres) = unsync_channel(None);
 
     spawn_local(async move {
+        let send_ingress = send_ingres;
+        let mut peers_send = HashMap::new();
+        let mut peers_recv = StreamMap::new();
+
         loop {
-            let (stream, peer_addr) = if let Ok((stream, _)) = listener.accept().await {
-                if let Ok(peer_addr) = stream.peer_addr() {
-                    (stream, peer_addr)
-                } else {
-                    continue;
+            // Calling methods in a loop, futures must be cancel-safe.
+            select! {
+                biased;
```

Comment:
It will be helpful to document why `biased` is necessary.

Reply:
What happens if you don't put `biased`?

Reply:
I think I forgot to unresolve this conversation.

Explaining what happens if `biased` is removed from here will be helpful. It seems to be one word: doing a lot of heavy lifting.

Reply:
I have no idea what biased means :-)

Reply:
Added more detailed comments for your reading pleasure :P